platform/wayland/cog-platform-wl.c

```diff
@@ -757,14 +961,30 @@ registry_global (void               *data,
         wl_data.subcompositor = wl_registry_bind(registry, name, &wl_subcompositor_interface, version);
     } else if (strcmp(interface, wl_shell_interface.name) == 0) {
         wl_data.shell = wl_registry_bind(registry, name, &wl_shell_interface, version);
+        win_data.shell_functions->enter_fullscreen = &wl_shell_enter_fullscreen;
```

Comment:
Rather than setting each function, it might be worthwhile just having each set of functions pre-defined, so something like:
```
   static const shell_operations wayland_shell_operations = {
     .enter_fullscreen = &wl_shell_enter_fullscreen,
     .exit_fullscreen = &wl_shell_exit_fullscreen,
     ...
   };
   
   ...
   
     win_data.shell_functions = &wayland_shell_operations
```
As it is, `win_data.shell_functions` is a pointer and this is referencing invalid memory as I don't see anywhere where `shell_functions` is set.